### ISSUE TITLE
Chrome/Chromium support

### ISSUE THREAD

**Issue by HLFH** [Synzvato/decentraleyes#1]
Hi,

It would be cool to have Chrome/Chromium support for decentraleyes.

**Comment by Synzvato** [comment]
Hi, thanks for your suggestion!

In order to keep the codebase as flexible (and the goals as focused) as possible, builds for other web browsers, such as Chrome, are not under active development at this point.

The choice for Firefox was made based on the Mozilla Foundation's organizational structure (non-profit) and its [manifesto](https://www.mozilla.org/en-US/about/manifesto), which is very much in line with what Decentraleyes stands for as a project.

However, once Mozilla officially drops its support for XPCOM and XUL based add-ons, Decentraleyes will be ported to WebExtension (a technology designed to facilitate cross-browser compatibility). I believe that would be the right time to have a closer look at this matter.

**Comment by HLFH** [comment]
@Synzvato I understand but IMHO "Mozilla Foundation's organizational structure (non-profit) and its manifesto" are sadly now complete bullshit. I understand we still choose at first Mozilla for its nice past. Thanks for developing decentraleyes and good luck for WebExtension port when it will be the time.

**Comment by cowlicks** [comment]
@Synzvato WebExtension will be stable in FF later this year. Do you have a roadmap for porting in mind?

**Comment by Synzvato** [comment]
@cowlicks Since Decentraleyes leans on a couple of relatively exotic APIs, it all depends on what's possible with Mozilla's implementation of ```WebExtensions``` once it turns stable.

All I can really say at this point is that I'll try to get _(or keep)_ essential APIs on the team's radar and will start porting the add-on to ```WebExtensions``` once I have sufficient confidence in the implementation.

**Comment by Synzvato** [comment]
In reply to #65 by @AMP648

> There is clearly a demand for Chromium support. I know you said there's no plans for it but I think you should reconsider this request.

Actually, there are concrete plans for creating a more universal version of the add-on. Expect some details once Mozilla stabilizes their new ```WebExtensions``` API implementation.

> There are plenty of open source Chrome browsers that follow Mozillas principles. Please consider porting this to chromre.

I would personally recommend Mozilla Firefox over Chromium-based browsers _(including more privacy-minded solutions like SRWare Iron, for reasons such as [these](https://github.com/nylira/prism-break/issues/169))_ to those concerned about spying. That said, I'm trying my very best to bring this concept to those who are on Chromium or Opera.

I'm actively working on it, and will get back to this issue once I have some details.

**Comment by dlmetcalf** [comment]
Exciting news!  Thank you for this.  Quite a few of us use more than one browser, depending on where we are and for what, as Firefox still has some significant issues, e.g. see https://twitter.com/CopperheadSec/status/730691939532574720

**Comment by 9Morello** [comment]
Any news on this?

**Comment by Synzvato** [comment]
@9Morello Porting the add-on to ```WebExtensions``` is a blocking issue for its very next release. So, if all goes well, the next version will support Opera, Chromium, Chrome, and (possibly) Edge. I'm stuck waiting on an effort to bridge a small [feature gap](https://bugzilla.mozilla.org/show_bug.cgi?id=1256122), but things are starting to look up, it seems!

**Comment by Droyk** [comment]
anynews on this? what happened?

**Comment by Synzvato** [comment]
@Droyk While it's theoretically possible for a ```WebExtensions``` add-on to redirect requests to local resources, this has significant [security implications](https://github.com/mozilla/addons-linter/issues/1035#issuecomment-258878823). That's why I'm now first aiming for additions to the current ```WebExtensions``` APIs. Until then, it's impossible for anyone to release a secure port.

Despite the various setbacks, please know that porting Decentraleyes to ```WebExtensions``` is definitely going to happen, and that it has top priority. I will be sure to keep you posted.

**Comment by Droyk** [comment]
>please know that porting Decentraleyes to WebExtensions is definitely going to happen, and that it has top priority.

All I want to know is an ETA now plz don't tell me you don't have that and plz don't lie about this.

**Comment by dlmetcalf** [comment]
> All I want to know is an ETA now plz don't tell me you don't have that and plz don't lie about this.

Seriously @Droyk?!  It's a volunteer project.  If you want to make demands on a 'product' you paying nothing for,  then how about writing your own to make it on?

@Synzvato, thanks for whatever you get to do on it.

**Comment by Droyk** [comment]
>If you want to make demands

Looks like you guys got it wrong I am not making demands at all I just want to know ETA if that seems like a demand to you guys then forgive me guys I was totally wrong to ask you guys a simple ETA.

**Comment by HLFH** [comment]
@Droyk There is no ETA. It's an old issue. And it has been closed. We will talk about an ETA when a developer will make a PR, or when the repo owner @Synzvato decides to reopen this issue. It is not the case. We are right now into the brainstorming step, not into the making and decision process.

**Comment by Droyk** [comment]
ohh okay thanx for informing me that.

**Comment by Synzvato** [comment]
**Update:** I'm still trying to find out if the issues are exploitable in Chromium. Some doubts remain, but my stance on this might change in the near future, as I haven't been able to cook up any exploits so far.

**Comment by Droyk** [comment]
that's a good news hope to hear soon from you about releasing it.

**Comment by Synzvato** [comment]
Decentraleyes is now available, for Chromium-based browsers! As a side note, the Opera-release is finished too, but is currently awaiting moderation. Apologies for the wait, but we could not release while having doubts about various security aspects. All is looking good, though. Thanks for your patience!

**Download and install the package manually**
https://github.com/Synzvato/decentraleyes/releases/tag/v1.3.7

**Install the extension from the Chrome Web Store**
https://chrome.google.com/webstore/detail/decentraleyes/ldpochfccmkkmhdbclfhpagapcfdljkj

**Comment by Droyk** [comment]
At last thanx @Synzvato

**Comment by HLFH** [comment]
Thank you @Synzvato!

**Comment by javacoolme** [comment]
YES :DDDDDDD
Thanks @Synzvato 👍

**Comment by LumenReaper** [comment]
Runs well in Firefox. I'd like to know if the extension for Edge is still planned. 

Thanks!

**Comment by andmpel** [comment]
@LumenReaper
You might want to create a new issue for this.
